applications/GeoMechanicsApplication/custom_elements/README.md

```diff
@@ -0,0 +1,130 @@
+# Transient Thermal Element
+
+## Introduction
+The calculation of geothermal processes relies on a profound understanding of heat transport mechanisms within the Earth's subsurface. Central to this understanding is the heat transport equation. In this context, the heat transport equation becomes a linchpin for designing sustainable and efficient geothermal systems.
+
+## Governing Equations
+The primary governing equation for heat transport in geothermal applications is the heat convection-conduction equation, which describes how heat flows through a porous medium. 
+
+$$ \left(n S \rho^w c^w + (1- n) \rho^s c^s \right) \frac{\partial T}{\partial t} = -\rho^w c^w q_i \frac{\partial T}{\partial x_i} + \frac{\partial}{\partial x_i} \left( D_{ij} \frac{\partial T}{\partial x_j} \right) \qquad \text{on} \quad \Omega $$
+
+where,
+
+- $c^w$    = specific heat capacity liquid phase $\mathrm{[J/kg ^{\circ}C]}$
+- $c^s$    = specific heat capacity solid phase  $\mathrm{[J/kg ^{\circ}C]}$
+- $D_{ij}$ = hydrodynamic thermal dispersion  $\mathrm{[W/m ^{\circ}C]}$
+- $T$      = temperature  $\mathrm{[ ^{\circ}C]}$
+- $\rho^s$ = density solid phase $\mathrm{[kg/m^3]}$ 
+- $\rho^w$ = water density $\mathrm{[kg/m^3]}$ 
+- $q$      = specific discharge $\mathrm{[m/s]}$ 
+- $S$      = degree of saturation $\mathrm{[-]}$
+- $n$      = porosity $\mathrm{[-]}$
+
+The hydrodynamic thermal dispersion is defined as:
+
+$$ D_{ij}= nS \lambda^w \delta_{ij} + \left(1-n\right) \lambda_{ij}^s + c^w \rho^w \left( (\alpha_l - \alpha_t) \frac{q_i q_j}{q} + \delta_{ij} \alpha_t q \right) $$
+
+where
+
+- $\alpha_l$    = longitudinal dispersivity $\mathrm{[m]}$
+- $\alpha_t$    = transverse dispersivity $\mathrm{[m]}$
+- $\delta_{ij}$ = Kronecker delta $\mathrm{[-]}$
+- $\lambda^w$   = thermal conductivity water $\mathrm{[W/m ^{\circ}C]}$
+- $\lambda^s$   = thermal conductivity solid matrix $\mathrm{[W/m ^{\circ}C]}$
+
+In the absence of ground water flow, these equations are simplified to,
+
+$$ \left(n S \rho^w c^w + (1- n) \rho^s c^s \right) \frac{\partial T}{\partial t} = \frac{\partial}{\partial x_i} \left( D_{ij} \frac{\partial T}{\partial x_j} \right) \qquad \text{on} \quad \Omega $$
+
+$$ D_{ij}= nS \lambda^w \delta_{ij} + \left(1-n\right) \lambda_{ij}^s $$
+
+## Boundary Conditions
+
+### Dirichlet Boundary Condition
+$$ T = \overline T \qquad \text{on} \quad \Gamma_{1}^T $$
+
+where $\overline T$ $\mathrm{\left[ ^{\circ}C \right]}$ is a prescribed temperature 
+
+### Neumann Boundary Condition
+$$ D_{ij} \frac{\partial T}{\partial x_j} n_i = \overline{f} \qquad \text{on} \quad \Gamma_{2}^T $$
+
+where $\overline f$ $\mathrm{\left[ W/m^2 \right]}$ is a prescribed conductive heat flux.
```

Comment:
Please could you explain why here f is just a **conductive**' heat flux and below q is **convective-conductive** heat flux?